mavros/src/plugins/param.cpp

```diff
@@ -86,30 +87,37 @@ class Parameter {
 		case enum_value(MT::INT8):
 			int_tmp = uv.param_int8;
 			param_value = int_tmp;
+			param_type = MT::INT8;
 			break;
 		case enum_value(MT::UINT8):
 			int_tmp = uv.param_uint8;
 			param_value = int_tmp;
+			param_type = MT::UINT8;
 			break;
 		case enum_value(MT::INT16):
 			int_tmp = uv.param_int16;
 			param_value = int_tmp;
+			param_type = MT::INT16;
 			break;
 		case enum_value(MT::UINT16):
 			int_tmp = uv.param_uint16;
 			param_value = int_tmp;
+			param_type = MT::UINT16;
 			break;
 		case enum_value(MT::INT32):
 			int_tmp = uv.param_int32;
 			param_value = int_tmp;
+			param_type = MT::INT32;
 			break;
 		case enum_value(MT::UINT32):
 			int_tmp = uv.param_uint32;
 			param_value = int_tmp;
+			param_type = MT::UINT32;
 			break;
 		case enum_value(MT::REAL32):
 			float_tmp = uv.param_float;
 			param_value = float_tmp;
+			param_type = MT::REAL32;
```

Comment:
No, you shouldn't change generated code. Cog script above is the source.
But i do not see a reason to save param type. You can get type of param_value.